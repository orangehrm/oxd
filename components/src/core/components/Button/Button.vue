<!--
/*
 * This file is part of OrangeHRM Inc
 *
 * Copyright (C) 2020 onwards OrangeHRM Inc
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */
-->

<template>
  <button type="button" :class="classes" :style="style">
    <slot name="icon">
      <oxd-icon v-if="iconName" :name="iconName" class="oxd-button-icon" />
    </slot>
    {{ label }}
    <slot name="iconRight">
      <oxd-icon
        v-if="iconRightName"
        :name="iconRightName"
        class="oxd-button-icon"
      />
    </slot>
  </button>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {
  ButtonSize,
  ButtonType,
  SIZES,
  SIZE_MEDIUM,
  TYPES,
  TYPE_MAIN,
} from './types';
import Icon from '@orangehrm/oxd/core/components/Icon/Icon.vue';

export default defineComponent({
  name: 'oxd-button',

  components: {
    'oxd-icon': Icon,
  },

  props: {
    label: {
      type: String,
      required: true,
    },
    displayType: {
      type: String,
      default: TYPE_MAIN,
      validator: function(value: ButtonType) {
        return TYPES.indexOf(value) !== -1;
      },
    },
    size: {
      type: String,
      default: SIZE_MEDIUM,
      validator: function(value: ButtonSize) {
        return SIZES.indexOf(value) !== -1;
      },
    },
    style: {
      type: Object,
    },
    iconName: {
      type: String,
    },
    iconRightName: {
      type: String,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-button': true,
        [`oxd-button--${this.size}`]: true,
        [`oxd-button--${this.displayType}`]: true,
      };
    },
  },
});
</script>

<style src="./button.scss" lang="scss" scoped></style>
