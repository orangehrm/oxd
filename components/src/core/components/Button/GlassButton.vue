<!--
/*
 * This file is part of OrangeHRM Inc
 *
 * Copyright (C) 2020 onwards OrangeHRM Inc
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */
-->

<template>
  <button type="button" :class="btnClasses" :disabled="disabled">
    <span v-if="icon" :class="iconClasses"></span>
    <slot v-else name="icon"></slot>
    {{ label }}
  </button>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {GlassIconType, GLASS_ICON_TYPES} from './types';

export default defineComponent({
  name: 'OxdGlassButton',

  props: {
    label: {
      type: String,
      required: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    icon: {
      type: String,
      required: false,
      validator: function (value: GlassIconType) {
        return GLASS_ICON_TYPES.indexOf(value) !== -1;
      },
    },
  },

  computed: {
    btnClasses(): object {
      return {
        'oxd-glass-button': true,
        [`oxd-glass-button--disabled`]: this.disabled,
      };
    },
    iconClasses(): object {
      return {
        'oxd-glass-button-icon': true,
        'oxd-glass-button-icon--disabled': this.disabled,
        [`oxd-glass-button-icon--${this.icon}`]: this.icon,
      };
    },
  },
});
</script>

<style src="./glass-button.scss" lang="scss" scoped></style>
