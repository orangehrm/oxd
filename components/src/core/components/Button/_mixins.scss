@mixin oxd-button-color($background-color) {
  color: $oxd-button-font-color;
  background-color: $background-color;
}

@mixin oxd-button-solid($type: 'main') {
  @include oxd-button-base;
  $oxd-button: map-get($oxd-solid-buttons, $type);

  @include oxd-button-color(map-get($oxd-button, 'active'));

  &:focus {
    @include oxd-button-color(map-get($oxd-button, 'focus'));
  }

  &:hover {
    @include oxd-button-color(map-get($oxd-button, 'hover'));
  }

  &:active {
    @include oxd-button-color(map-get($oxd-button, 'pressed'));
  }

  &:disabled {
    @include oxd-button-color(map-get($oxd-button, 'inactive'));
  }
}

@mixin oxd-button-label($type: 'default') {
  @include oxd-button-base;
  $oxd-button: map-get($oxd-label-buttons, $type);

  color: map-get($oxd-button, 'font');
  background-color: map-get($oxd-button, 'active');

  @if ($type == 'default' or $type == 'text') {
    &:focus {
      background-color: map-get($oxd-button, 'focus');
    }
    &:hover {
      background-color: map-get($oxd-button, 'hover');
    }
  } @else {
    &:focus {
      background-color: map-get($oxd-button, 'focus');
    }
    &:hover {
      background-color: map-get($oxd-button, 'hover');
    }
  }

  &:active {
    background-color: map-get($oxd-button, 'pressed');
  }

  &:disabled {
    @if ($type== 'text') {
      color: map-get($oxd-button, 'font-inactive');
    } @else {
      color: $oxd-button-font-color;
    }

    background-color: map-get($oxd-button, 'inactive');
  }
}

@mixin oxd-button-label-feedback($type: 'default') {
  @include oxd-button-base;
  $oxd-button: map-get($oxd-label-feedback-buttons, $type);

  color: map-get($oxd-button, 'font');
  background-color: map-get($oxd-button, 'active');

  @if ($type == 'default' or $type == 'text') {
    &:focus {
      background-color: map-get($oxd-button, 'focus');
    }
    &:hover {
      background-color: map-get($oxd-button, 'hover');
    }
  } @else {
    &:focus {
      background-color: map-get($oxd-button, 'focus');
    }
    &:hover {
      background-color: map-get($oxd-button, 'hover');
    }
  }

  &:focus {
    background-color: map-get($oxd-button, 'focus');
  }
  &:hover,
  &:focus,
  &:active {
    color: $oxd-button-font-color;
  }

  &:disabled {
    @if ($type== 'text') {
      color: map-get($oxd-button, 'font-inactive');
    } @else {
      color: $oxd-button-font-color;
    }

    background-color: map-get($oxd-button, 'inactive');
  }
}

@mixin oxd-button-color-ghost($color, $hover: false) {
  border: $oxd-button-border--ghost;

  @if ($hover) {
    color: $oxd-button-font-color;
    background-color: $color;
  } @else {
    color: $color;
    background-color: transparent;
  }
}

@mixin oxd-button-ghost($type: 'default') {
  @include oxd-button-base;
  $oxd-button: map-get($oxd-ghost-buttons, $type);

  @include oxd-button-color-ghost(map-get($oxd-button, 'active'));

  @if ($type == 'default' or $type == 'tool') {
    &:focus {
      @include oxd-button-color-ghost(map-get($oxd-button, 'focus'));
    }
    &:hover {
      @include oxd-button-color-ghost(map-get($oxd-button, 'hover'));
    }
  } @else {
    &:focus {
      @include oxd-button-color-ghost(map-get($oxd-button, 'focus'), true);
    }
    &:hover {
      @include oxd-button-color-ghost(map-get($oxd-button, 'hover'), true);
    }
  }

  &:active {
    @include oxd-button-color-ghost(map-get($oxd-button, 'pressed'));
  }

  &:disabled {
    @include oxd-button-color-ghost(map-get($oxd-button, 'inactive'));
  }
}

@mixin oxd-button-base {
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
