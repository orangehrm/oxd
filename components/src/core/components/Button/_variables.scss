// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

$oxd-button-min-width: 100px !default;
$oxd-button-horizontal-padding: 32px !default;
$oxd-button-vertical-padding--large: 16px !default;
$oxd-button-vertical-padding--medium: 8px !default;
$oxd-button-vertical-padding--small: 4px !default;
$oxd-button-font-size--large: 14px !default;
$oxd-button-font-size--medium: 14px !default;
$oxd-button-font-size--small: 12px !default;
$oxd-button-font-weight: 600 !default;
$oxd-button-font-family: $oxd-font-family !default;
$oxd-button-font-color: $oxd-white-color !default;

/**
 * Solid Buttons
**/

/* Main */
$oxd-button-color--main: $oxd-primary-one-color !default;
$oxd-button-color--main-pressed: darken($oxd-button-color--main, 5.1%) !default;
$oxd-button-color--main-hover: lighten($oxd-button-color--main, 4.9%) !default;
$oxd-button-color--main-focus: $oxd-button-color--main-hover !default;
$oxd-button-color--main-inactive: $oxd-interface-gray-lighten-2-color !default;

/* Secondary */
$oxd-button-color--secondary: $oxd-secondary-four-color !default;
$oxd-button-color--secondary-pressed: darken(
  $oxd-button-color--secondary,
  5%
) !default;
$oxd-button-color--secondary-hover: lighten(
  $oxd-button-color--secondary,
  5%
) !default;
$oxd-button-color--secondary-focus: $oxd-button-color--secondary-hover !default;
$oxd-button-color--secondary-inactive: $oxd-interface-gray-lighten-2-color !default;

/* Info */
$oxd-button-color--info: $oxd-feedback-info-color !default;
$oxd-button-color--info-pressed: darken($oxd-button-color--info, 5%) !default;
$oxd-button-color--info-hover: lighten($oxd-button-color--info, 5%) !default;
$oxd-button-color--info-focus: $oxd-button-color--info-hover !default;
$oxd-button-color--info-inactive: $oxd-interface-gray-lighten-2-color !default;

/* Danger */
$oxd-button-color--danger: $oxd-feedback-danger-color !default;
$oxd-button-color--danger-pressed: darken(
  $oxd-button-color--danger,
  5%
) !default;
$oxd-button-color--danger-hover: lighten(
  $oxd-button-color--danger,
  5%
) !default;
$oxd-button-color--danger-focus: $oxd-button-color--danger-hover !default;
$oxd-button-color--danger-inactive: $oxd-interface-gray-lighten-2-color !default;

/* Warning */
$oxd-button-color--warn: $oxd-feedback-warn-color !default;
$oxd-button-color--warn-pressed: darken($oxd-button-color--warn, 5%) !default;
$oxd-button-color--warn-hover: lighten($oxd-button-color--warn, 5%) !default;
$oxd-button-color--warn-focus: $oxd-button-color--warn-hover !default;
$oxd-button-color--warn-inactive: $oxd-interface-gray-lighten-2-color !default;

/* Success */
$oxd-button-color--success: $oxd-feedback-success-color !default;
$oxd-button-color--success-pressed: darken(
  $oxd-button-color--success,
  5%
) !default;
$oxd-button-color--success-hover: lighten(
  $oxd-button-color--success,
  5%
) !default;
$oxd-button-color--success-focus: $oxd-button-color--success-hover !default;
$oxd-button-color--success-inactive: $oxd-interface-gray-lighten-2-color !default;

$oxd-solid-buttons: (
  main: (
    active: $oxd-button-color--main,
    pressed: $oxd-button-color--main-pressed,
    focus: $oxd-button-color--main-focus,
    hover: $oxd-button-color--main-hover,
    inactive: $oxd-button-color--main-inactive,
  ),
  secondary: (
    active: $oxd-button-color--secondary,
    pressed: $oxd-button-color--secondary-pressed,
    focus: $oxd-button-color--secondary-focus,
    hover: $oxd-button-color--secondary-hover,
    inactive: $oxd-button-color--secondary-inactive,
  ),
  info: (
    active: $oxd-button-color--info,
    pressed: $oxd-button-color--info-pressed,
    focus: $oxd-button-color--info-focus,
    hover: $oxd-button-color--info-hover,
    inactive: $oxd-button-color--info-inactive,
  ),
  danger: (
    active: $oxd-button-color--danger,
    pressed: $oxd-button-color--danger-pressed,
    focus: $oxd-button-color--danger-focus,
    hover: $oxd-button-color--danger-hover,
    inactive: $oxd-button-color--danger-inactive,
  ),
  warn: (
    active: $oxd-button-color--warn,
    pressed: $oxd-button-color--warn-pressed,
    focus: $oxd-button-color--warn-focus,
    hover: $oxd-button-color--warn-hover,
    inactive: $oxd-button-color--warn-inactive,
  ),
  success: (
    active: $oxd-button-color--success,
    pressed: $oxd-button-color--success-pressed,
    focus: $oxd-button-color--success-focus,
    hover: $oxd-button-color--success-hover,
    inactive: $oxd-button-color--success-inactive,
  ),
);

/**
 * Ghost Button
**/
$oxd-button-color--ghost: $oxd-secondary-four-color !default;
$oxd-button-border--ghost: $oxd-border !default;
$oxd-button-color--ghost-pressed: darken($oxd-button-color--ghost, 5%) !default;
$oxd-button-color--ghost-hover: lighten($oxd-button-color--ghost, 5%) !default;
$oxd-button-color--ghost-focus: $oxd-button-color--ghost-hover !default;
$oxd-button-color--ghost-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Info */
$oxd-button-color--ghost-info: $oxd-feedback-info-color !default;
$oxd-button-color--ghost-info-pressed: darken(
  $oxd-button-color--ghost-info,
  5%
) !default;
$oxd-button-color--ghost-info-hover: $oxd-button-color--ghost-info !default;
$oxd-button-color--ghost-info-focus: $oxd-button-color--ghost-info-hover !default;
$oxd-button-color--ghost-info-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Danger */
$oxd-button-color--ghost-danger: $oxd-feedback-danger-color !default;
$oxd-button-color--ghost-danger-pressed: darken(
  $oxd-button-color--ghost-danger,
  5%
) !default;
$oxd-button-color--ghost-danger-hover: $oxd-button-color--ghost-danger !default;
$oxd-button-color--ghost-danger-focus: $oxd-button-color--ghost-danger-hover !default;
$oxd-button-color--ghost-danger-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Warning */
$oxd-button-color--ghost-warn: $oxd-feedback-warn-color !default;
$oxd-button-color--ghost-warn-pressed: darken(
  $oxd-button-color--ghost-warn,
  5%
) !default;
$oxd-button-color--ghost-warn-hover: $oxd-button-color--ghost-warn !default;
$oxd-button-color--ghost-warn-focus: $oxd-button-color--ghost-warn-hover !default;
$oxd-button-color--ghost-warn-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Success */
$oxd-button-color--ghost-success: $oxd-feedback-success-color !default;
$oxd-button-color--ghost-success-pressed: darken(
  $oxd-button-color--ghost-success,
  5%
) !default;
$oxd-button-color--ghost-success-hover: $oxd-button-color--ghost-success !default;
$oxd-button-color--ghost-success-focus: $oxd-button-color--ghost-success-hover !default;
$oxd-button-color--ghost-success-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Tool */
$oxd-button-color--ghost-tool: $oxd-interface-gray-color !default;
$oxd-button-color--ghost-tool-pressed: $oxd-interface-gray-darken-1-color !default;
$oxd-button-color--ghost-tool-hover: $oxd-interface-gray-darken-1-color !default;
$oxd-button-color--ghost-tool-focus: $oxd-interface-gray-darken-1-color !default;
$oxd-button-color--ghost-tool-inactive: $oxd-interface-gray-lighten-1-color !default;

$oxd-ghost-buttons: (
  default: (
    active: $oxd-button-color--ghost,
    pressed: $oxd-button-color--ghost-pressed,
    focus: $oxd-button-color--ghost-focus,
    hover: $oxd-button-color--ghost-hover,
    inactive: $oxd-button-color--ghost-inactive,
  ),
  info: (
    active: $oxd-button-color--ghost-info,
    pressed: $oxd-button-color--ghost-info-pressed,
    focus: $oxd-button-color--ghost-info-focus,
    hover: $oxd-button-color--ghost-info-hover,
    inactive: $oxd-button-color--ghost-info-inactive,
  ),
  danger: (
    active: $oxd-button-color--ghost-danger,
    pressed: $oxd-button-color--ghost-danger-pressed,
    focus: $oxd-button-color--ghost-danger-focus,
    hover: $oxd-button-color--ghost-danger-hover,
    inactive: $oxd-button-color--ghost-danger-inactive,
  ),
  warn: (
    active: $oxd-button-color--ghost-warn,
    pressed: $oxd-button-color--ghost-warn-pressed,
    focus: $oxd-button-color--ghost-warn-focus,
    hover: $oxd-button-color--ghost-warn-hover,
    inactive: $oxd-button-color--ghost-warn-inactive,
  ),
  success: (
    active: $oxd-button-color--ghost-success,
    pressed: $oxd-button-color--ghost-success-pressed,
    focus: $oxd-button-color--ghost-success-focus,
    hover: $oxd-button-color--ghost-success-hover,
    inactive: $oxd-button-color--ghost-success-inactive,
  ),
  tool: (
    active: $oxd-button-color--ghost-tool,
    pressed: $oxd-button-color--ghost-tool-pressed,
    focus: $oxd-button-color--ghost-tool-focus,
    hover: $oxd-button-color--ghost-tool-hover,
    inactive: $oxd-button-color--ghost-tool-inactive,
  ),
);

/**
 * Label Button
**/
$oxd-button-font-color--label: #68a61d !default;
$oxd-button-color--label: rgba($oxd-button-font-color--label, 0.1) !default;
$oxd-button-color--label-pressed: rgba(
  $oxd-button-font-color--label,
  0.2
) !default;
$oxd-button-color--label-hover: rgba(
  $oxd-button-font-color--label,
  0.15
) !default;
$oxd-button-color--label-focus: $oxd-button-color--label-hover !default;
$oxd-button-color--label-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Info */
$oxd-button-font-color--label-info: $oxd-feedback-info-color !default;
$oxd-button-color--label-info: rgba($oxd-feedback-info-color, 0.1) !default;
$oxd-button-color--label-info-pressed: rgba(
  $oxd-feedback-info-color,
  0.2
) !default;
$oxd-button-color--label-info-hover: $oxd-button-font-color--label-info !default;
$oxd-button-color--label-info-focus: $oxd-button-color--label-info-hover !default;
$oxd-button-color--label-info-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Danger */
$oxd-button-font-color--label-danger: $oxd-feedback-danger-color !default;
$oxd-button-color--label-danger: rgba($oxd-feedback-danger-color, 0.1) !default;
$oxd-button-color--label-danger-pressed: rgba(
  $oxd-feedback-danger-color,
  0.2
) !default;
$oxd-button-color--label-danger-hover: $oxd-button-font-color--label-danger !default;
$oxd-button-color--label-danger-focus: $oxd-button-color--label-danger-hover !default;
$oxd-button-color--label-danger-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Warning */
$oxd-button-font-color--label-warn: $oxd-feedback-warn-color !default;
$oxd-button-color--label-warn: rgba($oxd-feedback-warn-color, 0.1) !default;
$oxd-button-color--label-warn-pressed: rgba(
  $oxd-feedback-warn-color,
  0.2
) !default;
$oxd-button-color--label-warn-hover: $oxd-button-font-color--label-warn !default;
$oxd-button-color--label-warn-focus: $oxd-button-color--label-warn-hover !default;
$oxd-button-color--label-warn-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Success */
$oxd-button-font-color--label-success: $oxd-feedback-success-color !default;
$oxd-button-color--label-success: rgba(
  $oxd-feedback-success-color,
  0.1
) !default;
$oxd-button-color--label-success-pressed: rgba(
  $oxd-feedback-success-color,
  0.2
) !default;
$oxd-button-color--label-success-hover: $oxd-button-font-color--label-success !default;
$oxd-button-color--label-success-focus: $oxd-button-color--label-success-hover !default;
$oxd-button-color--label-success-inactive: $oxd-interface-gray-lighten-1-color !default;

/* Text button */
$oxd-button-font-color--label-text: $oxd-interface-gray-darken-1-color !default;
$oxd-button-color--label-text: rgba(
  $oxd-button-font-color--label-text,
  0.1
) !default;
$oxd-button-color--label-text-pressed: rgba(
  $oxd-button-font-color--label-text,
  0.2
) !default;
$oxd-button-color--label-text-hover: rgba(
  $oxd-button-font-color--label-text,
  0.15
) !default;
$oxd-button-color--label-text-focus: $oxd-button-color--label-text-hover !default;
$oxd-button-color--label-text-inactive: #f6f7f9 !default;
$oxd-button-font-color--label-text-inactive: $oxd-interface-gray-lighten-2-color !default;

$oxd-label-buttons: (
  default: (
    font: $oxd-button-font-color--label,
    active: $oxd-button-color--label,
    pressed: $oxd-button-color--label-pressed,
    focus: $oxd-button-color--label-focus,
    hover: $oxd-button-color--label-hover,
    inactive: $oxd-button-color--label-inactive,
  ),
  info: (
    font: $oxd-button-font-color--label-info,
    active: $oxd-button-color--label-info,
    pressed: $oxd-button-color--label-info-pressed,
    focus: $oxd-button-color--label-info-focus,
    hover: $oxd-button-color--label-info-hover,
    inactive: $oxd-button-color--label-info-inactive,
  ),
  danger: (
    font: $oxd-button-font-color--label-danger,
    active: $oxd-button-color--label-danger,
    pressed: $oxd-button-color--label-danger-pressed,
    focus: $oxd-button-color--label-danger-focus,
    hover: $oxd-button-color--label-danger-hover,
    inactive: $oxd-button-color--label-danger-inactive,
  ),
  warn: (
    font: $oxd-button-font-color--label-warn,
    active: $oxd-button-color--label-warn,
    pressed: $oxd-button-color--label-warn-pressed,
    focus: $oxd-button-color--label-warn-focus,
    hover: $oxd-button-color--label-warn-hover,
    inactive: $oxd-button-color--label-warn-inactive,
  ),
  success: (
    font: $oxd-button-font-color--label-success,
    active: $oxd-button-color--label-success,
    pressed: $oxd-button-color--label-success-pressed,
    focus: $oxd-button-color--label-success-focus,
    hover: $oxd-button-color--label-success-hover,
    inactive: $oxd-button-color--label-success-inactive,
  ),
  text: (
    font: $oxd-button-font-color--label-text,
    active: $oxd-button-color--label-text,
    pressed: $oxd-button-color--label-text-pressed,
    focus: $oxd-button-color--label-text-focus,
    hover: $oxd-button-color--label-text-hover,
    inactive: $oxd-button-color--label-text-inactive,
    font-inactive: $oxd-button-font-color--label-text-inactive,
  ),
);

/**
 * Icon Button
**/
$oxd-icon-button-font-size: 18px !default;
$oxd-icon-button-font-weight: 400 !default;
$oxd-icon-button-font-color: $oxd-interface-gray-darken-1-color !default;
$oxd-icon-button-bg-color: rgba($oxd-icon-button-font-color, 0.1) !default;
$oxd-icon-button--pressed: rgba($oxd-icon-button-font-color, 0.2) !default;
$oxd-icon-button--hover: rgba($oxd-icon-button-font-color, 0.15) !default;
$oxd-icon-button--disabled: #f6f7f9 !default;
$oxd-icon-button-font-color--disabled: $oxd-interface-gray-lighten-2-color !default;
$oxd-icon-button-size: 36px;
$oxd-icon-button-padding: 6px;

/**
 * List Button
**/
$oxd-list-button-font-color: $oxd-interface-gray-darken-1-color !default;
$oxd-list-button--pressed: $oxd-primary-one-color !default;
$oxd-list-button--hover: $oxd-list-button--pressed !default;
