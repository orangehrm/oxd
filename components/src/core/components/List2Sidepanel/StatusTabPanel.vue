<template>
  <div class="oxd-status-tab-panel-header">
    <slot name="header" :collapsed="collapsed"></slot>
    <oxd-button
      v-if="!$slots['header']"
      size="long"
      flow="right"
      iconSize="xx-small"
      :class="mainButtonClasses"
      :displayType="mainButtonType"
      :iconName="mainButtonIconName"
      :disabled="mainButtonDisabled"
      :tooltip="$vt(mainButtonLabel)"
      :label="!collapsed ? $vt(mainButtonLabel) : ''"
      @click="onClickMainButton"
    />
  </div>
  <oxd-divider></oxd-divider>
  <div class="oxd-status-tab-panel-body">
    <slot name="body" :collapsed="collapsed"></slot>
    <ul
      :class="{
        'oxd-status-tab-panel-filters': true,
        '--padtop': $slots['body'],
      }"
    >
      <li
        v-for="filter in filterTabs"
        :key="filter"
        :class="{
          'oxd-status-tab-panel-filter': true,
          '--selected': filter.selected,
          '--collapsed': collapsed,
        }"
        @click.stop="onClickFilter(filter)"
      >
        <oxd-text v-if="!collapsed" class="oxd-status-tab-panel-filter-name">
          {{ $vt(filter.name) }}
        </oxd-text>
        <oxd-text
          flow="right"
          class="oxd-status-tab-panel-filter-count"
          :tooltip="collapsed ? $vt(filter.name) : ''"
          :style="{color: filter.color, background: filter.backgroundColor}"
        >
          {{ filter.count }}
        </oxd-text>
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import {Filter} from './types';
import translateMixin from '../../../mixins/translate';
import {computed, defineComponent, PropType} from 'vue';
import Text from '@orangehrm/oxd/core/components/Text/Text.vue';
import Divider from '@orangehrm/oxd/core/components/Divider/Divider.vue';
import OxdButton from '@orangehrm/oxd/core/components/Button/Button.vue';

export default defineComponent({
  inheritAttrs: false,

  name: 'oxd-status-tab-panel',

  mixins: [translateMixin],

  props: {
    collapsed: {
      type: Boolean,
      default: false,
    },
    mainButtonLabel: {
      type: String,
      default: 'Main Button',
    },
    mainButtonType: {
      type: String,
      default: 'secondary',
    },
    mainButtonIconName: {
      type: String,
      default: 'oxd-add',
    },
    mainButtonDisabled: {
      type: Boolean,
      default: false,
    },
    filters: {
      type: Array as PropType<Filter[]>,
      default: () => [],
    },
  },

  components: {
    'oxd-text': Text,
    'oxd-divider': Divider,
    'oxd-button': OxdButton,
  },

  emits: ['clickMainButton', 'filter'],

  setup(props, context) {
    const filterTabs = computed(() => {
      return props.filters.map(filter => {
        return {
          ...filter,
          color: filter.color,
          backgroundColor: filter.backgroundColor ?? '',
          count: String(filter.count ?? '0').substring(0, 6),
        };
      });
    });

    const mainButtonClasses = computed(() => ({
      'oxd-status-tab-panel-mainbutton': true,
      '--collapsed': props.collapsed,
    }));

    const onClickFilter = (filter: Filter) => {
      context.emit('filter', filter);
    };

    const onClickMainButton = ($event: Event) => {
      context.emit('clickMainButton', $event);
    };

    return {
      filterTabs,
      onClickFilter,
      onClickMainButton,
      mainButtonClasses,
    };
  },
});
</script>

<style lang="scss" src="./status-tab-panel.scss" scoped></style>
