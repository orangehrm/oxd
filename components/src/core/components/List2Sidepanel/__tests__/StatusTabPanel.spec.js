import {mount} from '@vue/test-utils';
import Button from '@orangehrm/oxd/core/components/Button/Button.vue';
import StatusTabPanel from '@orangehrm/oxd/core/components/List2Sidepanel/StatusTabPanel.vue';

const filterStub = [
  {
    count: '10',
    selected: false,
    name: 'In Progress',
    color: 'grey',
    backgroundColor: '#fff',
  },
  {
    count: '7',
    selected: true,
    name: 'Shortlisted',
    color: '#f6f5fb',
    backgroundColor: '#eb0910',
  },
];

describe('List2 > Sidepanel > StatusTabPanel.vue', () => {
  it('renders StatusTabPanel', () => {
    const wrapper = mount(StatusTabPanel);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders StatusTabPanel with filters', () => {
    const wrapper = mount(StatusTabPanel, {
      props: {
        filters: [...filterStub],
      },
    });
    expect(
      wrapper.findAll('.oxd-status-tab-panel-filter').length,
    ).toStrictEqual(2);
    expect(
      wrapper.findAll('.oxd-status-tab-panel-filter-name').map(el => el.text()),
    ).toStrictEqual(['In Progress', 'Shortlisted']);
    expect(
      wrapper
        .findAll('.oxd-status-tab-panel-filter-count')
        .map(el => el.text()),
    ).toStrictEqual(['10', '7']);
  });
  it('emits click event on click main button', async () => {
    const wrapper = mount(StatusTabPanel, {
      props: {
        filters: [...filterStub],
      },
    });
    wrapper.findComponent(Button).trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('clickMainButton')).toBeTruthy();
  });
  it('emits filter event on click filter element', async () => {
    const wrapper = mount(StatusTabPanel, {
      props: {
        filters: [...filterStub],
      },
    });
    wrapper.findAll('li')[0].trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('filter')).toStrictEqual([
      filterStub.filter((_, i) => i === 0),
    ]);
  });
});
