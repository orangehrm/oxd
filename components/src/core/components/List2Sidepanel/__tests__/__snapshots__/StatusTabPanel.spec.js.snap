// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List2 > Sidepanel > StatusTabPanel.vue renders StatusTabPanel 1`] = `
<div class="oxd-status-tab-panel-header"><button type="button" class="oxd-button oxd-button--long oxd-button--secondary with-icon oxd-status-tab-panel-mainbutton" tooltip="Main Button" flow="right"><i class="oxd-svg-icon oxd-svg-icon--xx-small oxd-button-icon"><svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 459.3 459.3" style="enable-background:new 0 0 459.3 459.3;" xml:space="preserve">
        <style type="text/css">
          .st0 {
            fill: currentColor;
          }
        </style>
        <g>
          <path class="st0" d="M459.3,229.7c0,22.2-18,40.2-40.2,40.2H269.9v149.3c0,22.2-18,40.2-40.2,40.2c-11.1,0-21.1-4.5-28.4-11.8 c-7.3-7.3-11.8-17.3-11.8-28.4l0-149.3H40.2c-11.1,0-21.1-4.5-28.4-11.8C4.5,250.8,0,240.8,0,229.7c0-22.2,18-40.2,40.2-40.2h149.3 V40.2c0-22.2,18-40.2,40.2-40.2c22.2,0,40.2,18,40.2,40.2v149.3h149.3C441.3,189.5,459.3,207.5,459.3,229.7z"></path>
        </g>
      </svg></i>
    <div class="oxd-button-label-wrapper">Main Button</div>
    <!--v-if-->
  </button></div>
<div>
  <hr class="oxd-divider-horizontal" role="separator" aria-orientation="horizontal">
  <!--v-if-->
</div>
<div class="oxd-status-tab-panel-body">
  <ul class="oxd-status-tab-panel-filters"></ul>
</div>
`;
